futures-util/src/io/take.rs

```diff
@@ -0,0 +1,35 @@
+use futures_core::task::{Context, Poll};
+use futures_io::AsyncRead;
+use std::io;
+use std::pin::Pin;
+
+/// Future for the [`take`](super::AsyncReadExt::take) method.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Take<R: Unpin> {
+    inner: R,
+    limit: u64,
+}
+
+impl<R: AsyncRead + Unpin> Take<R> {
```

Comment:
We can probably add some methods [such as `limit`, `set_limit`](https://doc.rust-lang.org/nightly/std/io/struct.Take.html#methods). This can happen in this PR or follow-up work.

Reply:
Those are trivial additions, I'll go ahead and add them now.

Reply:
Take also has into_inner, get_ref, and get_mut - I can add those as well.

Reply:
> Take also has into_inner, get_ref, and get_mut - I can add those as well.

I think they are useful.